### PR TITLE
Mepts 149 implement logs of what records have been updated

### PR DESCRIPTION
**Pull request by cfaife**

### CODE REVIEW

File: src/main/resources/pdiresources/jobs/migration.kjb [hunk 1]
```diff
@@ -573,66 +572,12 @@
   <hops>
     <hop>
       <from>Start</from>
-      <to>harmonize program workflow states</to>
```
Comment (ynurmahomed) [inline src/main/resources/pdiresources/jobs/migration.kjb:576]: Migration is running only `program` and `form`. Please add the other ones.

File: src/main/resources/pdiresources/encounter_type/log-for-encounter-type-references.ktr [hunk 2]
```diff
@@ -0,0 +1,1288 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (ynurmahomed) [inline src/main/resources/pdiresources/encounter_type/log-for-encounter-type-references.ktr:1]: This transformation is never executed.

File: src/main/resources/pdiresources/form/log-for-form-references.ktr [hunk 3]
```diff
@@ -0,0 +1,1709 @@
...
+, creator old_creator
+, date_created old_date_created
+, changed_by old_changed_by
+, date_changed old_date_changed
+, retired old_retired
+, retired_by old_retired_by
+, date_retired old_date_retired
+, retired_reason old_retired_reason
+, uuid old_uuid
+, updated old_updated
```
Comment (ynurmahomed) [inline src/main/resources/pdiresources/form/log-for-form-references.ktr]: `updated` column is not added to metadata table, this causes the transformation to fail. Similar for other metadata.

**Review by ynurmahomed — CHANGES_REQUESTED** [review]
Hi @cfaife the logic for the logs looks ok. But there are some issues, please have a look in the comments.

**Comment by ynurmahomed** [comment]
@cfaife the file `form-data.sql` was added, but it is not referenced anywhere in the project.

File: src/main/resources/pdiresources/encounter_type/log-for-encounter-type-references.ktr (hunk 2 above)
Comment (cfaife) [reply src/main/resources/pdiresources/encounter_type/log-for-encounter-type-references.ktr:1]: done

File: src/main/resources/pdiresources/form/log-for-form-references.ktr (hunk 3 above)
Comment (cfaife) [reply src/main/resources/pdiresources/form/log-for-form-references.ktr]: done

**Review by ynurmahomed — APPROVED** [review]

**Pull request merged**